MediaBrowser.Providers/Tmdb/Movies/TmdbSearch.cs

```diff
@@ -86,36 +100,35 @@ private async Task<IEnumerable<RemoteSearchResult>> GetSearchResults(ItemLookupI
                 }
             }
 
+            // Ideally retrying alternatives should be done outside the search
+            // provider so that the retry logic can be common for all search
+            // providers
             if (results.Count == 0)
             {
-                // try with dot and _ turned to space
-                var originalName = name;
-
-                name = name.Replace(",", " ");
-                name = name.Replace(".", " ");
-                name = name.Replace("_", " ");
-                name = name.Replace("-", " ");
-                name = name.Replace("!", " ");
-                name = name.Replace("?", " ");
-
-                var parenthIndex = name.IndexOf('(');
-                if (parenthIndex != -1)
-                {
-                    name = name.Substring(0, parenthIndex);
-                }
+                name = parsedName.Name;
+
+                // Remove things enclosed in []{}() etc
+                name = _cleanEnclosed.Replace(name, string.Empty);
+
+                // Replace sequences of non-word characters with space
+                name = _cleanNonWord.Replace(name, " ");
 
+                // Clean based on common stop words / tokens
+                name = _cleanStopWords.Replace(name, string.Empty);
+
+                // Trim whitespace
                 name = name.Trim();
 
                 // Search again if the new name is different
-                if (!string.Equals(name, originalName))
+                if (!string.Equals(name, parsedName.Name) && !string.IsNullOrWhiteSpace(name))
```

Comment:
this check is wrong. you should be comparing to previous search term (which was already partially cleaned).

Reply:
agree it could lead to needlessly querying with same query twice